projects/plugins/jetpack/extensions/blocks/subscriptions/save.js

```diff
@@ -128,13 +128,31 @@ export default function Save( { className, attributes } ) {
 		success_message: successMessage,
 	};
 
+	/**
+	 * Apply HTML encoding for special characters inside shortcode attributes: <, >, " and '.
+	 *
+	 * @see https://codex.wordpress.org/Shortcode_API#Attributes
+	 * @param {string} value - Value to encode.
+	 * @returns {string} Encoded value.
+	 */
+	const encodeValueForShortcodeAttribute = value => {
+		return value
+			.replace( /</g, '&lt;' )
+			.replace( />/g, '&gt;' )
+			.replace( /"/g, '&quot;' )
+			.replace( /'/g, '&#039;' );
```

Comment:
It breaks in some cases unless we also include `[` and `]`.
```suggestion
			.replace( /'/g, '&#039;' )
			.replace( /\[/g, '&#091;' )
			.replace( /\]/g, '&#093;' );
```
And as mentioned earlier, we may want to include NBSP and ZWSP to avoid them being mangled.
```suggestion
			.replace( /'/g, '&#039;' )
			.replace( /\[/g, '&#091;' )
			.replace( /\]/g, '&#093;' )
			.replace( /\u00a0/g, '&nbsp;' )
			.replace( /\u200b/g, '&#x200b;' );
```
Looks like whether `&` should be encoded or not is tricky, as the value passed in for `submitButtonText` already has ampersands (and `<`) escaped while the value for `successMessage` does not. Hooray for consistency? I guess we should leave it out.

Reply:
@anomiex, thanks for finding those missing cases. I've added square brackets, nbsp, and zwsp to the chain. It looks like the square brackets were the biggest offender here, breaking the post entirely.